File: src/main/java/ge/edu/sangu/drive/cluster/Node.java
package ge.edu.sangu.drive.cluster;

import org.jetbrains.annotations.NotNull;

/**
 * @author irakli on 4/19/2015.
 */
public class Node implements Comparable<Node> {

	public String id;

	@Override
	public int compareTo(@NotNull Node other) {
		return this.id.compareTo(other.id);
	}
}

File: src/main/java/ge/edu/sangu/drive/cluster/Node.kt
package ge.edu.sangu.drive.cluster

import java.util.*

/**
 * @author irakli on 4/19/2015.
 */
public class Node (var uuid : String = UUID.randomUUID().toString()) : Comparable<Node> {

    public var id: String = uuid
        get() = $id
        set(value) {$id = value}

    override fun compareTo(other: Node): Int {
        return this.id.compareTo(other.id)
    }
}
